lib/vector-buffers/src/topology/channel/sender.rs

```diff
@@ -322,13 +307,20 @@ impl<T: Bufferable> Sink<T> for BufferSender<T> {
                 if result.is_ok() {
                     *this.base_flush = true;
 
-                    if let Some(item_size) = item_size {
+                    // NOTE: This is potentially a smol lie because we haven't yet actually accepted
+                    // the item yet in some cases, due to needing to buffer it temporarily to ensure
+                    // we have room for it, since `poll_ready` gives us no way to know the item size
+                    // ahead of time.
```

Comment:
This is maybe something we could build into our version of a `reserve` function to replace `poll_ready`.

Reply:
Yeah, agreed.  Actual `async fn`s will let us write much more self-evident code.